test/pretty_print_test.exs

```diff
@@ -394,6 +394,19 @@ defmodule Erlex.Test.PretyPrintTest do
     assert pretty_printed == expected_output
   end
 
+  test "when clauses in contracts pretty print appropriately" do
+    input = ~S"""
+    (Atom,Encoding) -> binary() when Atom :: atom(), Encoding :: 'latin1' | 'utf8'
+    """
+
+    pretty_printed = Erlex.pretty_print(input)
+
+    expected_output =
+      "(:Atom, :Encoding) :: binary() when Atom: atom(), Encoding: :latin1 | :utf8"
```

Comment:
`Atom` and `Encoding` are variables, not atoms. I think the output should be:

`"(atom, encoding) :: binary() when atom :: atom(), encoding :: :latin1 | :utf8"`

Reply:
That's a deeper issue this exposed is slightly out of scope of this fix.  I need to figure out how to detect Erlang terms as opposed to Elixir ones and handle them specially.